File: server/iih_pub/iih.ci_pub/src/main/java/iih/ci/mr/nu/obstetrics/antenatalassess/d/AntenatalassessAggDO.java
package iih.ci.mr.nu.obstetrics.antenatalassess.d;

import xap.sys.appfw.orm.model.agg.BaseAggDO;
import xap.mw.core.data.BaseDO;
import xap.mw.coreitf.i.IAggDesc;
import xap.mw.coreitf.i.IBaseDO;
import iih.ci.mr.nu.obstetrics.antenatalassess.d.desc.AntenatalassessAggDODesc;



/**
 * 产科护理记录单(产后、术后)
 */
public class AntenatalassessAggDO extends BaseAggDO {
	private static final long serialVersionUID = 1L;

	@Override
	public AntenAssDO getParentDO() {
		return ((AntenAssDO) super.getParentDO());
	}

	public void setParentDO(AntenAssDO headDO) {
		setParent(headDO);
	}

	public AntenNurBserRecDO[] getAntenNurBserRecDO() {
		IBaseDO[] dos = getChildren(AntenNurBserRecDO.class);
		if(dos==null) return null;
		AntenNurBserRecDO[] result=new AntenNurBserRecDO[dos.length];
		for(int i=0;i<dos.length;i++){
			result[i]=(AntenNurBserRecDO)dos[i];
		}
		return result;
	}
	
	public void setAntenNurBserRecDO(AntenNurBserRecDO[] dos) {
		setChildren(AntenNurBserRecDO.class, dos);
	}

	@Override
	public IAggDesc getAggDesc() {
		AntenatalassessAggDODesc desc = new AntenatalassessAggDODesc();

		return desc;
	}
	
	 public  BaseDO createParentDO() {
         return new AntenAssDO();
     }

     public  BaseDO createChildDO(String clzName) {
	     if (clzName.equals("iih.ci.mr.nu.obstetrics.antenatalassess.d.AntenNurBserRecDO")) {
             return new AntenNurBserRecDO();
         }
         return null;
     }
}

File: client/iih_stub/iih.ci_stub/iih.ci.mr/nu/obstetrics/antenatalassess/d/AntenatalassessAggDO.cs
using System;
using xap.mw.core.data;
using xap.mw.core.meta;

namespace iih.ci.mr.nu.obstetrics.antenatalassess.d
{
	/// <summary>
	/// 产科护理记录单(产后、术后)DO
	/// </summary>
	public class AntenatalassessAggDO : BaseAggDO
	{	
	    private static string ANTENASSDO_ANTENNURBSERRECDO="iih.ci.mr.nu.obstetrics.antenatalassess.d.AntenNurBserRecDO";

        public AntenatalassessAggDO() {
            this.setParentDO(new AntenAssDO());
        }

  	    public AntenAssDO getParentDO() {
		    return ((AntenAssDO) base.ParentDO);
	    }      
	    
	    public void setParentDO(AntenAssDO headDO) {
		    base.ParentDO=headDO;
	    }
	    
	    public AntenNurBserRecDO[] getAntenNurBserRecDO() {		    
		    
		    BaseDO[] dos=getChildren(Type.GetType(ANTENASSDO_ANTENNURBSERRECDO));
		    if (dos == null || dos.Length==0){return null;}
            AntenNurBserRecDO[] r = new AntenNurBserRecDO[dos.Length];
            for (int i = 0; i < dos.Length;i++ ){
                r[i] = dos[i] as AntenNurBserRecDO;
            }
	        return r;
		    
	    }
	    
	    public void setAntenNurBserRecDO(AntenNurBserRecDO[] dos) {
            if (dos == null || dos.Length == 0) return;
            setChildren(dos[0].GetType(), dos);
	    }


	    public override BaseDO createParentDO()
	    {
	        return new AntenAssDO();
	    }

        public override BaseDO createChildDO(string clzName)
	    {
            if (ANTENASSDO_ANTENNURBSERRECDO == clzName)
	        {
                return new AntenNurBserRecDO();
	        }
	        return null;
	    }
	    
	    /// <summary>
        /// 获取子类全路径名数组
        /// </summary>
        /// <returns></returns>
        public override string[] getChildClassNames(){
            return  new string[]{"iih.ci.mr.nu.obstetrics.antenatalassess.d.AntenNurBserRecDO"};
        }
	    public new static AggMeta GetMeta()
	    {
	        AggMeta meta = new AggMeta();
	        meta.OneToOneDOs = new string[]{};
	        return meta;
	    }
	    
	}
}
